graphql/schema/mutation.graphql

```diff
@@ -63,18 +63,18 @@ type Mutation {
   createProject(project: CreateProjectInput! @requireProjectAdmin, requestS3Creds: Boolean): Project! 
   copyProject(project: CopyProjectInput! @requireProjectAdmin, requestS3Creds: Boolean): Project! 
-  deactivateStepbackTask(projectId: String! @requireProjectAccess(permission: SETTINGS, access: EDIT), buildVariantName: String!, taskName: String!): Boolean!
-  defaultSectionToRepo(projectId: String! @requireProjectAccess(permission: SETTINGS, access: EDIT), section: ProjectSettingsSection!): String
-  deleteProject(projectId: String! @requireProjectAdmin): Boolean!
-  detachProjectFromRepo(projectId: String! @requireProjectAccess(permission: SETTINGS, access: EDIT)): Project!
-  forceRepotrackerRun(projectId: String! @requireProjectAccess(permission: SETTINGS, access: EDIT)): Boolean!
-  promoteVarsToRepo(projectId: String! @requireProjectAccess(permission: SETTINGS, access: EDIT), varNames: [String!]!): Boolean!
-  saveProjectSettingsForSection(projectSettings: ProjectSettingsInput, section: ProjectSettingsSection!): ProjectSettings! # Has directive on ProjectSettingsInput.
-  saveRepoSettingsForSection(repoSettings: RepoSettingsInput, section: ProjectSettingsSection!): RepoSettings! # Has directive on RepoSettingsInput.
+  deactivateStepbackTask(projectId: String @deprecated(reason: "use opts.projectIdentifier instead") @requireProjectAccess(permission: SETTINGS, access: EDIT), buildVariantName: String @deprecated(reason: "use opts.buildVariantName instead"), taskName: String @deprecated(reason: "use opts.taskName instead"), opts: DeactivateStepbackTaskInput): Boolean!
+  defaultSectionToRepo(projectId: String @deprecated(reason: "use opts.projectIdentifier instead") @requireProjectAccess(permission: SETTINGS, access: EDIT), section: ProjectSettingsSection @deprecated(reason: "use opts.section instead"), opts: DefaultSectionToRepoInput): String
+  deleteProject(projectId: String @deprecated(reason: "use projectIdentifier instead") @requireProjectAdmin, projectIdentifier: String @requireProjectAdmin): Boolean!
+  detachProjectFromRepo(projectId: String @deprecated(reason: "use projectIdentifier instead") @requireProjectAccess(permission: SETTINGS, access: EDIT), projectIdentifier: String @requireProjectAccess(permission: SETTINGS, access: EDIT) ): Project!
+  forceRepotrackerRun(projectId: String @deprecated(reason: "use projectIdentifier instead") @requireProjectAccess(permission: SETTINGS, access: EDIT), projectIdentifier: String @requireProjectAccess(permission: SETTINGS, access: EDIT)): Boolean!
+  promoteVarsToRepo(projectId: String @deprecated(reason: "use opts.projectIdentifier instead") @requireProjectAccess(permission: SETTINGS, access: EDIT), varNames: [String!] @deprecated(reason: "use opts.varNames instead"), opts: PromoteVarsToRepoInput): Boolean!
```

Comment:
this might be a silly question haha but does the directive get ignored when the field is `null`? So if `projectId = null`, it won't throw an error that we don't have access to or can't operate on project `null`?

Reply:
hmm, I think that is the behavior I'm seeing. I would expect everything to fail due to this line if the value is null. https://github.com/evergreen-ci/evergreen/blob/09623203eca2078b277b5c314857de60d4cbf0fd/rest/data/middleware.go#L157